backend/src/hatchling/builders/config.py

```diff
@@ -785,6 +785,11 @@ def load_vcs_exclusion_patterns(self) -> list[str]:
                     if glob_mode:
                         patterns.append(line)
 
+        # validate project root is not excluded by vcs
+        exclude_spec = pathspec.GitIgnoreSpec.from_lines(patterns)
+        if exclude_spec.match_file(self.root):
+            return []
```

Comment:
```suggestion
            return patterns
```

Reply:
What's the reason for this change? AFAIU it would make this PR do nothing effectively as the entire point is to disable ignore patterns entirely if the root path matches the patterns(which indicates the project is not being tracked by git.